routes/index.tsx

```diff
@@ -52,7 +52,16 @@ export const handler: Handlers<HomePageData, State> = {
 export default function HomePage(props: PageProps<HomePageData>) {
   return (
     <>
-      <Head href={props.url.href} />
+      <Head href={props.url.href}>
+        <script
+          dangerouslySetInnerHTML={{
+            __html: `
+            window.isDark = localStorage.theme === "dark" || (!("theme" in localStorage) && window.matchMedia("(prefers-color-scheme: dark)").matches)
+            document.documentElement.classList[window.isDark ? 'add' : 'remove']("dark");
+            `,
+          }}
+        />
```

Comment:
I believe this script is unnecessary if we just want to support the system's theme preference.